src/layers/ZoomLayer.ts

```
@@ -60,17 +86,73 @@ export class ZoomLayer extends OptionalLayer {
     this.handleZoom({ x: [extentXStart, extentXEnd], y: [extentYStart, extentYEnd] }, layerArgs);
   };
 
+  private handleBrushMove = (event: d3.D3BrushEvent<Point>, layerArgs: LayerArgs) => {
+    if (!event.selection) return;
+    this.overlay!.style("display", "");
+    const [[x0, y0], [x1, y1]] = this.processSelection(event, layerArgs);
+    this.selectionMask!
+      .attr("x", Math.min(x0, x1))
+      .attr("y", Math.min(y0, y1))
+      .attr("width", Math.abs(x0 - x1))
+      .attr("height", Math.abs(y0 - y1));
+  };
+
   draw = (layerArgs: LayerArgs) => {
     const { width, height, margin } = layerArgs.bounds;
+    const svg = layerArgs.coreLayers[LayerType.Svg];
     
     // brushX allows the user to click and draw a rectangle that will
     // select a particular x interval and it will then fire an end event
     const d3Brush = d3.brush<Point>()
       .extent([[margin.left, margin.top], [width - margin.right, height - margin.bottom]]);
     const brushLayer = layerArgs.coreLayers[LayerType.BaseLayer].append("g")
-      .attr("id", layerArgs.getHtmlId(LayerType.Zoom))
+      .attr("id", `${layerArgs.getHtmlId(LayerType.Zoom)}-brush`)
+      // we hide the default brush layer and implement our own because
+      // we would like additional behaviour such as visually showing a
+      // 1D zoom if the selection has similar x or y coordinates
+      .style("opacity", 0)
```

The "default brush layer" is the `d3Brush`? And if you didn't hide it here, it would show the default box on drag?

---

yep exactly